config.md

```diff
@@ -317,6 +320,10 @@ Annotations are key-value maps.
 }
 ```
 
+## Extensibility
+The `annotations` property MAY be used as an extensibility point to include additional information that is not defined as part of this specification.
+Excluding the `annotations` property, implementations that are reading/processing this configuration file MUST generate an error if they encounter an unkown property.
```

Comment:
On Sun, Jun 26, 2016 at 05:37:21PM -0700, Doug Davis wrote:
> > +## Extensibility
> > +The `annotations` property MAY be used as an extensibility point to include additional information that is not defined as part of this specification.
> > +Excluding the `annotations` property, implementations that are reading/processing this configuration file MUST generate an error if they encounter an unkown property.
>
> I'd prefer we either ignore or fault unknown properties - no reason
> to open up interoperability issues with SHOULDs/MAYs/etc...

It's hard to know whether a property should be optional or not without
details of the property itself.  For example, runC feels like the
version check is unnecessary [1], and ccon feels like the platform
check is unnecessary [2].  In the absence of detail about the property
in question, the conservative course is to error out (so I'm in favor
of SHOULDing that approach).  But “the runtime doesn't know what that
property is for” doesn't mean “the caller doesn't know what that
property is for”, and I'd rather allow users to be able to say “don't
worry about unrecognized properties, they're not important for this
container” (so I'm not in favor of MUSTing that approach).  And a
SHOULD would avoid name migration issues as new features are
standardized [3].

On the other hand, I'm generally in favor of for moving non-standard
things out of the config (e.g. hooks.json, if we drop hooks [4]), and
a MUST here would certainly help with that ;).

[1]: https://github.com/opencontainers/runc/pull/521
[2]: https://github.com/wking/ccon/tree/v0.4.0#configuration
[3]: https://tools.ietf.org/html/rfc6648#appendix-B
[4]: https://groups.google.com/a/opencontainers.org/d/msg/dev/Y7p6YW8zr4s/NtiVw9p2AwAJ
     Subject: Re: Hooks and all-in-one operation
     Date: Thu, 9 Jun 2016 14:50:49 -0700
     Message-ID: <20160609215049.GK8801@odin.tremily.us>